gapic-generator-java/DEVELOPMENT.md

```diff
@@ -155,6 +155,32 @@ To generate a production GAPIC API:
    ```
    and then rebuild gapic-generator-java (`mvn clean install`).
 
+## Debugging the Plugin under googleapis with local gapic-generator-java (end-to-end debugging)
+
+To debug the local generator starting from protobuf compiler plugin
+
+1. Add [Remote JVM debug configuration](https://www.jetbrains.com/help/idea/tutorial-remote-debug.html) and breakpoints for the local generator in Intellij.
+
+2. In [googleapis](https://github.com/googleapis/googleapis) root directory, run the following commands:
+   1. Set `JVM_DEBUG_PORT` environment variable.
+       ```shell
+       export JVM_DEBUG_PORT=5005
+       ```
+   2. Run the command to build a client library, e.g., `java-monitoring`, with `--subcommands` flag to output commands spawned by `bazel build`.
+      ```shell
+      bazel build --subcommands //google/monitoring/v3:google-cloud-monitoring-v3-java
```

Comment:
If you set JVM_DEBUG_PORT first, this command stops.

Reply:
It doesn't stop in my environment. 

In fact, environment variable set in `bazel build` has no effect on the spawned subcommand, otherwise we don't need `--subcommands`.

Reply:
That's interesting observation. Then the order doesn't matter.